In [133]:
import numpy as np
import sklearn 
import pandas as pd

In [134]:
df=pd.read_csv("student.csv",sep=';') 

In [135]:
df.shape

(649, 33)

In [136]:
X=df[df.columns[30:32]]
Y=df[df.columns[32]]

In [137]:
X_train, X_test, y_train, y_test = sklearn.model_selection.train_test_split(X, Y, test_size=0.2, random_state=0)

In [138]:
regressor = sklearn.linear_model.LinearRegression()  
regressor.fit(X_train, y_train) 

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

In [139]:
regressor.intercept_
regressor.coef_

array([0.153563  , 0.91827126])

In [140]:
y_pred = regressor.predict(X_test)


In [141]:
print('Mean Absolute Error:', sklearn.metrics.mean_absolute_error(y_test, y_pred))  
print('Mean Squared Error:', sklearn.metrics.mean_squared_error(y_test, y_pred))  
print('Root Mean Squared Error:', np.sqrt(sklearn.metrics.mean_squared_error(y_test, y_pred)))


Mean Absolute Error: 0.7690621841478154
Mean Squared Error: 0.9857296674280447
Root Mean Squared Error: 0.9928391951509795


considering parmeters like study time , freetime ,health,etc

In [142]:
df.shape

(649, 33)

In [143]:
X1=df[["G1","G2","studytime","failures","goout","freetime","health","absences","age","traveltime"]]
Y1=df[df.columns[32]]

In [144]:
X1_train, X1_test, y1_train, y1_test = sklearn.model_selection.train_test_split(X1, Y1, test_size=0.2, random_state=0)

In [145]:
regressor1 = sklearn.linear_model.LinearRegression()  
regressor1.fit(X1_train, y1_train) 

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

In [146]:
regressor1.intercept_
regressor1.coef_

array([ 0.14606325,  0.90365999,  0.08548745, -0.24796639, -0.03010539,
       -0.03663212, -0.03843875,  0.02448719,  0.00747452,  0.1036208 ])

In [147]:
y1_pred = regressor1.predict(X1_test)


In [154]:
print('Mean Absolute Error:', sklearn.metrics.mean_absolute_error(y1_test, y1_pred))  
print('Mean Squared Error:', sklearn.metrics.mean_squared_error(y1_test, y1_pred))  
print('Root Mean Squared Error:', np.sqrt(sklearn.metrics.mean_squared_error(y1_test, y1_pred)))


Mean Absolute Error: 0.7445224178092672
Mean Squared Error: 0.9603257467108847
Root Mean Squared Error: 0.979962114936534


this suggests that marks of last two exams are enough to predict marks of students because adding extra parameters doesn't imporve performance by high factor.



parameters : marks of last two exams

Mean Absolute Error: 0.7690621841478154





parameters : marks of last two exams,study time , freetime ,health

Mean Absolute Error: 0.7445224178092672

author : dharmam savani
